### ISSUE TITLE
Remove the Task.Run from AzureKeyVaultTokenSigningService

### ISSUE THREAD

**Issue by omerlh** [MattCotterellNZ/IdentityServer3.Contrib.AzureKeyVaultTokenSigningService#9]
Hey
I've noticed that in `SignAsync` method there there is an async call using `Task.Run`. As far as I know (see this [SO](http://stackoverflow.com/a/29168586/4792970) post) this is not a good practice on a code that might run in webapi code. Also, the inner code, `AzureKeyVaultSignatureProvider` does call a true async function, and the current implementation hide it. It might be a good idea to create another interface which return Task for such operations.

**Comment by MattCotterellNZ** [comment]
Makes sense. We still need to keep the non-async method there since we need to override the `SignatureProvider`'s `Sign()` method, but there shouldn't be any harm in adding an additional `SignAsync()` (and `VerifyAsync()`) method too. Does that sound right?

**Comment by omerlh** [comment]
I don't see the benefit of using this interface, it might be better to create a dedicate interface for that...

**Comment by MattCotterellNZ** [comment]
I think this was partly to provide support for people who may not be using IdentityServer but still want to use parts of this library to sign JWTs using Azure Key Vault, but this is probably misguided and should probably be well out of scope (or in a different NuGet package altogether). I'll keep this in mind before release and look into it in a bit more depth, it may be appropriate to break away from this interface.
